pkg/apis/security/v1beta1/types.go

```diff
@@ -0,0 +1,191 @@
+	// this field. If this field is empty or missing, this rule matches all
+	// destinations.
+	// +optional
+	To []NetworkPolicyPeer `json:"to"`
+}
+
+// NetworkPolicyPeer describes the grouping selector of workloads.
+type NetworkPolicyPeer struct {
+	// IPBlock describes the IPAddresses/IPBlocks that is matched in to/from.
+	// IPBlock cannot be set as part of the AppliedTo field
+	// Cannot be set with any other selector.
+	// +optional
+	IPBlock *IPBlock `json:"ipBlock,omitempty"`
+	// Select Pods from NetworkPolicy's Namespace as workloads in
+	// AppliedTo/To/From fields. If set with NamespaceSelector, Pods are
+	// matched from Namespaces matched by the NamespaceSelector.
+	// Cannot be set with any other selector except NamespaceSelector.
+	// +optional
+	PodSelector *metav1.LabelSelector `json:"podSelector,omitempty"`
+	// Select all Pods from Namespaces matched by this selector, as
+	// workloads in To/From fields. If set with PodSelector,
+	// Pods are matched from Namespaces matched by the NamespaceSelector.
+	// Cannot be set with any other selector except PodSelector or
+	// ExternalEntitySelector.
+	// +optional
+	NamespaceSelector *metav1.LabelSelector `json:"namespaceSelector,omitempty"`
+	// Select ExternalEntities from NetworkPolicy's Namespace as workloads
+	// in AppliedTo/To/From fields. If set with NamespaceSelector,
+	// ExternalEntities are matched from Namespaces matched by the
+	// NamespaceSelector.
+	// Cannot be set with any other selector except NamespaceSelector.
+	ExternalEntitySelector *metav1.LabelSelector `json:"externalEntitySelector,omitempty"`
+}
+
+// IPBlock describes a particular CIDR (Ex. "192.168.1.1/24") that is allowed
+// or denied to/from the workloads matched by a Spec.AppliedTo.
+type IPBlock struct {
+	// CIDR is a string representing the IP Block
+	// Valid examples are "192.168.1.1/24".
+	CIDR string `json:"cidr"`
+}
+
+// NetworkPolicyPort describes the port and protocol to match in a rule.
+type NetworkPolicyPort struct {
+	// The protocol (TCP, UDP, or SCTP) which traffic must match.
+	// If not specified, this field defaults to TCP.
+	// +optional
+	Protocol *v1.Protocol `json:"protocol"`
+	// The port on the given protocol. This can either be a numerical
+	// or named port on a Pod. If this field is not provided, this
+	// matches all port names and numbers.
+	// TODO: extend it to include Port Range.
+	// +optional
+	Port *intstr.IntOrString `json:"port"`
+}
+
+// RuleAction describes the action to be applied on traffic matching a rule.
+type RuleAction string
+
+const (
+	// RuleActionAllow describes that rule matching traffic must be allowed.
+	RuleActionAllow RuleAction = "ALLOW"
+	// RuleActionDrop describes that rule matching traffic must be dropped.
+	RuleActionDrop RuleAction = "DROP"
+)
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type ExternalEntity struct {
```

Comment:
I thought ExternalNode is better for "agented" model, but maybe you want to cover agentless too, so chose ExternalEntity?

Reply:
Do you feel this is not specific to NetworkPolicy / security?

Reply:
i agree.. not specific to security.. will add it to a new group.. does "network" sound like a good group name? or "endpoint" group?

Reply:
Actually I am not confident on the naming. So if you have a better name, would like to hear.

Reply:
i think two other names in mind were externalName or externalHost